src/components/TextInput/index.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { memo, useCallback, useRef } from 'react'
+import styled from 'styled-components'
+
+const Input = styled.input<{ error?: boolean; fontSize?: string }>`
```

Comment:
Couldn't you use rebass text input here? I think it could simplify the css. Not really needed if this works

Reply:
Most of this is extracted from NumericalInput if i remember correctly.